packages/opentelemetry-core/src/context/propagation/TraceContextFormat.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SpanContext, Propagator } from '@opentelemetry/types';
+
+export const TRACE_PARENT_HEADER = 'traceparent';
+export const TRACE_STATE_HEADER = 'tracestate';
+// TODO: https://github.com/open-telemetry/opentelemetry-js/pull/60
+// Consider to import const from TraceOptions
+export const DEFAULT_OPTIONS = 0x0;
+const TRACE_PARENT_REGEX = /^[\da-f]{2}-[\da-f]{32}-[\da-f]{16}-[\da-f]{2}$/;
+const VERSION = '00';
+const traceIdFormat = /^[0-9a-f]{32}$/i;
+const spanIdFormat = /^[0-9a-f]{16}$/i;
+const invalidIdFormat = /^0+$/i;
+
+// TODO: Conside to move these interfaces in types.
+export interface HeaderGetter {
+  getHeader(name: string): string | string[] | undefined;
+}
+export interface HeaderSetter {
+  setHeader(name: string, value: string): void;
+}
+
+/**
+ * Propagates {@link SpanContext} through Trace Context format propagation.
+ *
+ * Based on the Trace Context specification:
+ * https://www.w3.org/TR/trace-context/
+ */
+export class TraceContextFormat implements Propagator {
+  inject(spanContext: SpanContext, format: string, carrier: HeaderSetter) {
```

Comment:
Some more context on why implementing with `HeaderSetter` works even though the interface has `unknown`: [Implementing interfaces but being more specific - bug or feature of typescript ?](https://github.com/Microsoft/TypeScript/issues/16735) that gives some nice explanation of this. Tl;dr - it is a feature!

Reply:
Thanks for the suggestion and sharing the useful link. I have updated the PR with required change.